src/igvfd/schemas/construct_library.json

```diff
@@ -0,0 +1,368 @@
+            "type": "array",
+            "minItems": 1,
+            "uniqueItems": true,
+            "items": {
+                "title": "Targeted Locus",
+                "type": "object",
+                "required": [
+                    "assembly",
+                    "chromosome",
+                    "start",
+                    "end"
+                ],
+                "additionalProperties": false,
+                "properties": {
+                    "assembly": {
+                        "title": "Mapping assembly",
+                        "description": "The genome assembly to which coordinates relate (e.g., GRCh38).",
+                        "type": "string",
+                        "enum": [
+                            "GRCh38",
+                            "hg19",
+                            "GRCm39",
+                            "mm10",
+                            "mm9"
+                        ]
+                    },
+                    "chromosome": {
+                        "title": "Chromosome",
+                        "description": "The number (or letter) designation for the chromosome, e.g. chr1 or chrX",
+                        "type": "string",
+                        "pattern": "^(chr[0-9A-Za-z]+)$"
+                    },
+                    "start": {
+                        "title": "Start",
+                        "description": "The 1-based, closed (inclusive) starting coordinate.",
+                        "type": "integer",
+                        "minimum": 0
+                    },
+                    "end": {
+                        "title": "End",
+                        "description": "The 1-based, closed (inclusive) ending coordinate.",
+                        "type": "integer",
+                        "minimum": 0
+                    }
+                }
+            }
+        },
+        "plasmid_map": {
+            "title": "Plasmid Map",
+            "description": "An image of the plasmid map used in this construct library.",
+            "type": "string",
+            "comment": "See document.json for a list of available identifiers.",
+            "linkTo": "Document"
+        },
+        "average_size": {
+            "title": "Average Size",
+            "description": "The average size of the inserts cloned into vectors in the library.",
+            "comment": "May not be submitted with size_range.",
+            "type": "number",
+            "minimum": 0
+        },
+        "size_range": {
+            "title": "Size Range",
+            "description": "The numerical range of the size of the inserts cloned in vectors in the library.",
+            "comment": "May not be submitted with average_size.",
+            "type": "string",
+            "pattern": "(^[0-9]+-[0-9]+$|^[<>][0-9]+$)"
+        },
+        "mpra_library_details": {
```

Comment:
should not this be just a property "MPRA library complexity" ? rather than object with this single property?

Reply:
unless we have some additional properties in mind it is hard to justify object? maybe directionality of inserts? or location of element in respect to the reporter gene? maybe reporter gene?

Reply:
As discussed, size of fragments (lower_ and upper_bound_insert_size, average_insert_size) are now in this sub-object. We expect to add something like reporter gene, directionality, other details of elements in the vector if labs want these details to be searchable/obvious on the page